hadoop-common-project/hadoop-common/src/main/resources/core-default.xml

```diff
@@ -1873,11 +1873,9 @@
 
 <property>
   <name>fs.s3a.committer.magic.enabled</name>
-  <value>false</value>
+  <value>true</value>
```

Comment:
with this change we will enable magic committer for all the components. Maybe point that out in the title of the issue.

Reply:
I also wonder if this (along with the documentation change) is required for this PR.

Reply:
I' will add to release notes. This *does not* enable the committer. All it does is say "this store has the consistency needed for the committer". 

Now, we could do a bit more in a separate patch as there's a bit more in settings I'd like to change (we ship in cloudera) where the fs.s3a.buffer.dir is set to something like `${env.LOCAL_DIR:${hadoop.tmp.dir}}/s3a` so that incomplete buffered writes (staging committer and blocks for normal files/magic uploads) are always cleaned up. I could make the core-default settings for this committer different

Reply:
gabor, disabling option is now in HADOOP-17483; #2637 a PR to completely remove the switch entirely. Now we have to decide: do we want that, or would we ever want to turn the s3a client into a dumb client which doesn't do "magic" when start writing under certain paths?